src/System.Linq/tests/ConcatTests.cs

```diff
@@ -421,5 +421,18 @@ public void GetEnumerableOfConcatCollectionChainFollowedByEnumerableNodeShouldBe
                 Assert.Equal(0xf00, en.Current);
             }
         }
+
+        [Fact]
+        public void CollectionInterleavedWithLazyEnumerables_ToArray_Regression()
```

Comment:
Nit: Do we need the "_Regression" suffix on these tests?  In a year it won't matter to someone reading the code whether we added a test proactively or in response to a bug.

Reply:
@stephentoub I personally added `_Regression` because someone might think "what a weird test. What is this supposed to cover?" / etc. in the future. Admittedly, it might be more helpful to remove the suffix and just link to the github issue in a comment:

```cs
        public void CollectionInterleavedWithLazyEnumerables_ToArray()
        {
            // This is a regression test for https://github.com/dotnet/corefx/issues/23680
            ...
        }
```

Reply:
> what a weird test. What is this supposed to cover?

That's why test naming is important, to qualify what the test is for.  Comments are good, too, of course.  But "_Regression" as a suffix is so moment-in-time.  If you were to follow true TDD, for example, everything would need to be suffixed with "_Regression" because the tests are all written before any code is and thus all fail initially.

Reply:
I'll add a comment with the bug URL.